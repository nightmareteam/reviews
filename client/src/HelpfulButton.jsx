/* eslint-disable indent */
import React from 'react';
//import styled from 'styled-components';
const styled = window.styled;

/**
 * Component for display of individual rating buttons
 * @param {Boolean} active - active state of the button
 * @param {Function} updateHelpfulness - function to update the state on click
 */
const HelpfulButton = (props) => (
      <Button active={props.active} onClick={(e) => props.updateHelpfulness(e, props.string.toLowerCase())}>{props.string}</Button>
 );

const Button = styled.span`
  border-radius: 2px;
  padding: 4px 6px 4px 6px;
  font-size: 12px;
  line-height: 20px;
  cursor: pointer;
  color: ${props => props.active ? 'white' : '#66c0f4' };
  border-color: transparent;
  display: inline;
  background: ${props => props.active ? '#36617c' : '#212c3d' };
  margin-left: 2px;
  margin-right: 2px;

  &:hover {
    background: #66c0f4;
    color: white;
  }
`;

export default HelpfulButton;