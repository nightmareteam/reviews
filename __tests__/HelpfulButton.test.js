import React from 'react';
import { shallow } from 'enzyme';
import HelpfulButton from '../client/src/HelpfulButton.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({
  updateHelpfulness: mockCallback,
  active: false
});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<HelpfulButton {...props} /> );
});

describe('<HelpfulButton /> rendering', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});