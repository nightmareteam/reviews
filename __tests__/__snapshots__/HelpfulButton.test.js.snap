// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelpfulButton /> rendering should render elements 1`] = `
<styled.span
  active={false}
  onClick={[Function]}
/>
`;
